<?xml version="1.0"?>
<launch>

    <!-- Launch Gazebo with underwater world -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="gui" value="true" />
        <arg name="paused" value="true"/>
        <arg name="world_name" value="$(find freefloating_gazebo)/world/underwater.world"/>
    </include>
  
    <!-- spawn in Gazebo with GUI -->
    <group ns="steering_auv">
        <param name="robot_description" command="$(find xacro)/xacro $(find freefloating_gazebo_demo)/urdf/steering_thrusters.xacro"/>
        <node name="spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-urdf -model steering_auv -param robot_description -x 0 -y 0 -z 1 -R 0 -P 0 -Y 0"/>

        <!-- Load controller configurations (vehicle and arm) from YAML file to parameter server -->
        <rosparam file="$(find freefloating_gazebo_demo)/config/steering_thrusters.yaml" command="load"/>

        <!-- Launch low-level PID control for arm 
             body control will not activate as we expect direct thruster control -->
        <node name="pid_control" pkg="freefloating_gazebo" type="pid_control" output="screen"/>
    </group>
    
    <!-- spawn terrain -->
    <param name="terrain_description" command="$(find xacro)/xacro $(find freefloating_gazebo_demo)/terrain/shipwreck/terrain.sdf"/>
    <node name="terrain_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-sdf -model terrain -param terrain_description -x 0 -y 0 -z 0 -R 0 -P 0 -Y 0"/>
   
</launch>
